### DISCUSSION TITLE
Is it possible to modify the matrix in ArchR or add external matrices to the project

### DISCUSSION THREAD

**Discussion by qingnanl** [GreenleafLab/ArchR#582] [Feature Requests]
Hi Developers,

Thanks for the great package. It is very smooth and versatile.

One thing I found is that I don't know how to modify the matrix generated by ArchR and store it back in the project or add external matrices to the project. For example, I might need to do some preprocessing such as batch correction or other normalization on the GeneScoreMatrix with other software and send it back to the ArchR project for downstream analysis. Is that possible or did I missed it in your tutorial?

Many thanks,

qingnanl

**Comment by rcorces** [comment]
As far as I know, this capability does not exist. It also seems difficult to implement. The matrices are actually stored in the Arrow files which complicates things. This does seem like it would be useful but @jgranja24 would have to weigh in on how difficult this would be to implement.

**Comment by qingnanl** [comment]
> As far as I know, this capability does not exist. It also seems difficult to implement. The matrices are actually stored in the Arrow files which complicates things. This does seem like it would be useful but @jgranja24 would have to weigh in on how difficult this would be to implement.

Thanks for the reply. I would still recommend you consider this possibility. I think for all the steps, you have put quite a state of the art methods in it, but it could be inevitable that some methods perform better than each other in a different context, and it is certainly impractical for you to get all of them implemented. 

I am also reading your code regarding "addGeneScoreMatrix" or adding other matrices add see whether it is possible to get something from it. Seems not easy for me lol. Anyway thanks again for the reply. The work is beautiful.

**Comment by rcorces** [comment]
What exactly is the downstream use case here? Do you want to display this new information as a UMAP, with cells colored by some updated value? If so, you can just add a new column to `cellColData` using values computed outside of ArchR. If you want a matrix to work natively with other ArchR functions, then that seems like a more difficult feature request that wont be implemented any time soon.

**Comment by qingnanl** [comment]
> What exactly is the downstream use case here? Do you want to display this new information as a UMAP, with cells colored by some updated value? If so, you can just add a new column to `cellColData` using values computed outside of ArchR. If you want a matrix to work natively with other ArchR functions, then that seems like a more difficult feature request that wont be implemented any time soon.

For example, what about batch correction on the gene score? Harmony is a good one but I won't say it is the best in all cases, plus it does not correct for the expression level itself. It might be useful for giving better RNA-ATAC integration and also gene-peak linking.

**Comment by xiaosuyu1997** [comment]
That is exactly a feature I'm looking for during my reproduction of someone else's work. I'm trying to import a Cicero gene activity score matrix into ArchR_project for downstream integration and label transfer analysis, and to give this matrix a customized name.

**Reply by urnotchris1** [reply]
Hi, I have the same problem as you. Have you solved it now?

**Reply by rcorces** [reply]
This feature is still not available. Sorry

**Reply by urnotchris1** [reply]
May I ask whether the result file generated by ArchR can be converted to CDS object ?

**Reply by rcorces** [reply]
What do you mean by CDS object?

**Reply by urnotchris1** [reply]
I want to calculate the Cicero gene activity score, this function takes cell_data_set (CDS) object as input. I wonder if input data can be generated directly from the result of ArchR

**Reply by rcorces** [reply]
No. Unfortunately there is no direct way to link between ArchR and Cicero. Though I dont believe there are many functionalities offered by Cicero that are not already offered by ArchR. Gene activity score estimation is highly customizable in ArchR and could be made to entirely mimic what is done in Cicero (though if you look at our Nature Genetics manuscript you will see that the gene score from Cicero is less accurate). And our coaccessibility linkage should be extremely similar in scope.

**Reply by urnotchris1** [reply]
It is appreciated for answering my question in your busy time!  I have used the getCoAccessibility() function and got the co-accessibility information, I am now looking for ways to use scATAC data to prediction enhancer as accurately as possible. Do you have any suggestions?

**Reply by rcorces** [reply]
I dont think there is much more than scATAC alone could tell you. Integrating with other data types (HiC, HiChIP, ChIP-seq) is probably the way to go.

**Reply by urnotchris1** [reply]
Many thanks for your time and additional explanations!!

**Comment by mairamirza** [comment]
Hi Developers, 
I want to load 10X scATAC-seq data ( matrix.mtx, features.tsv, and barcode.tsv) to ArchR to calculate co-accessibility. Is there any way to do that?

Thanks!

**Reply by rcorces** [reply]
No. the only acceptable inputs to ArchR are fragment files or bam files.

**Comment by qdong2023** [comment]
Hi @rcorces, I have a similar question. Is it possible to remove a matrix from the ArchR project? For example, there are 4 matrices in a project "GeneIntegrationMatrix" "GeneScoreMatrix" "PeakMatrix"  "TileMatrix", how to remove "GeneIntegrationMatrix"?
